### PR TITLE
Update to latest DBConnection

### PR DESCRIPTION
**Pull request by fishcakez**
WIP: awaiting DBConnection release

* Adds support for non-stream cursors
* Adds support for advance transaction handling

### CODE REVIEW

File: lib/postgrex/protocol.ex [hunk 1]
```diff
@@ -340,65 +330,75 @@ defmodule Postgrex.Protocol do
 
   @spec handle_begin(Keyword.t, state) ::
     {:ok, Postgrex.Result.t, state} |
-    {:error, Postgrex.Error.t, state} |
-    {:error | :disconnect, %RuntimeError{}, state} |
-    {:disconnect, %DBConnection.ConnectionError{}, state}
+    {DBConnecation.status, state} |
```
Comment (samphilipd) [inline lib/postgrex/protocol.ex]: Seems like a typo?

Comment (fishcakez) [inline lib/postgrex/protocol.ex]: Thanks fixed.

Comment (fishcakez) [inline lib/postgrex/protocol.ex]: Thanks fixed.

**Comment by fishcakez** [comment]
`max_rows` should be a `fetch` only option and not stored in cursor, the behaviour in this branch is confusing.

The `mode` should be stored in cursor so that when the cursor is deallocated the savepoint is cleaned up if option is not passed. This problem is a particular issue when copying from the database.

**Comment by josevalim** [comment]
@fishcakez I believe I have addressed the two concerns you had left. :)

**Comment by fishcakez** [comment]
@josevalim yes the changes are good 👍

**Pull request merged**